### PR TITLE
setup docker for project containerization

### PR DESCRIPTION
**Pull request by t4christ**
## What does this PR do?
     Sets up docker  environment on barefoot nomad project.
## Description of Task to be completed?
      Write dockerfile and docker-compose configuration file

## How should this be manually tested?
     pull the branch/clone the repo
     cd into the branch/repo
     install docker
     run docker-compose up

## What are the relevant pivotal tracker stories?
      [#167730626]

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -43,7 +43,7 @@ bower_components
 build/Release
 
 # Dependency directories
-node_modules/
+/node_modules
```
Comment (luqmanoop) [inline .gitignore:47]: Why was this changed from `node_modules/` to `/node_modules`?

Comment (t4christ) [reply .gitignore:47]: When i pulled develop i saw node modules folder active on my project. Not git ignored. When i made the change then the whole node modules folder was git ignored

**Review by chuksjoe — CHANGES_REQUESTED** [review]
Hello @t4christ, nice work bro. But I will advise you pull the latest merged content on develop branch, then confirm the root endpoint is accessible from the Docker container via a web browser.

**Comment by t4christ** [comment]
@chuksjoe thanks have done that.

**Comment by chuksjoe** [comment]
Hello @t4christ, please, can you shed some light on the nginx and why it's relevant.

**Comment by t4christ** [comment]
@chuksjoe it is meant for load balancing and serves as a request proxy to the server

**Review by danoseun — APPROVED** [review]

**Review by chuksjoe — APPROVED** [review]
Yet to confirm that docker works on my system. But with the integration setup done here the images and containers were successfully created on my computer.
Nice work @t4christ.

**Review by Kaytbode — APPROVED** [review]
@t4christ Well done!
